### PR TITLE
chore(votes): change parameters for propose_block vote

### PR DESCRIPTION
**Pull request by MatusKysel**
## Purpose

As we talked yesterday about this, we use wrong parameter for propose_block vote type. This PR is fixing it.
If we are ok with that we should prepare hardfork for testnet

### CODE REVIEW

File: libraries/core_libs/consensus/src/pbft/pbft_manager.cpp [hunk 1]
```diff
@@ -1126,9 +1150,10 @@ blk_hash_t PbftManager::calculateOrderHash(std::vector<DagBlock> const &dag_bloc
 std::pair<blk_hash_t, bool> PbftManager::proposeMyPbftBlock_() {
   auto round = getPbftRound();
   // In propose block, only use one vote for VRF sortition
-  VrfPbftSortition vrf_sortition(vrf_sk_, {propose_vote_type, round, step_});
+  VrfPbftSortition vrf_sortition(vrf_sk_, {propose_vote_type, round, 1});
```
Comment (JakubFornadel) [inline libraries/core_libs/consensus/src/pbft/pbft_manager.cpp:1153]: proposing is always only step 1 ?

File: libraries/core_libs/consensus/src/pbft/pbft_manager.cpp [hunk 2]
```diff
@@ -674,8 +677,29 @@ bool PbftManager::stateOperations_() {
...
+      if (v->getStep() == 1) {
+        // We need to handle propose_vote_type
+        try {
+          v->validate(1, getDposTotalAddressCount(), std::min(getDposTotalAddressCount(), COMMITTEE_SIZE));
+        } catch (const std::logic_error &e) {
+          LOG(log_wr_) << e.what();
+          return false;
+        }
+      } else {
+        try {
```
Comment (JakubFornadel) [inline libraries/core_libs/consensus/src/pbft/pbft_manager.cpp:692]: put it into single try catch block

File: libraries/core_libs/network/src/tarcap/stats/node_stats.cpp [hunk 3]
```diff
@@ -150,6 +151,7 @@ void NodeStats::logNodeStats() {
   LOG(log_nf_) << "PBFT chain size:                 " << local_chain_size;
   LOG(log_nf_) << "Current PBFT round:              " << local_pbft_round;
   LOG(log_nf_) << "DPOS total votes count:          " << local_dpos_total_votes_count;
+  LOG(log_nf_) << "DPOS total address count:        " << local_dpos_total_address_count;
```
Comment (JakubFornadel) [inline libraries/core_libs/network/src/tarcap/stats/node_stats.cpp:154]: addresses ?

File: libraries/core_libs/consensus/src/pbft/pbft_manager.cpp [hunk 4]
```diff
@@ -674,8 +677,29 @@ bool PbftManager::stateOperations_() {
   LOG(log_tr_) << "PBFT current step is " << step_;
 
   // Periodically verify unverified votes
-  vote_mgr_->verifyVotes(round, sortition_threshold_, getDposTotalVotesCount(),
-                         [this](auto const &addr) { return dposEligibleVoteCount_(addr); });
+  vote_mgr_->verifyVotes(round, [this](auto const &v) {
```
Comment (kstdl) [inline libraries/core_libs/consensus/src/pbft/pbft_manager.cpp:680]: maybe this is kind of tediousness from my side, but I think this is little easier to understand 
```cpp
  vote_mgr_->verifyVotes(round, [this](auto const &v) {
    auto dpos_votes_count = dposEligibleVoteCount_(v->getVoterAddr());
    if (!dpos_votes_count) {
      return false;
    }

    auto total_votes_count = getDposTotalVotesCount();
    auto threshold = sortition_threshold_;

    if (v->getStep() == 1) {
      // We need to handle propose_vote_type
      // verification differs for step 1
      dpos_votes_count = 1;
      total_votes_count = getDposTotalAddressCount();
      threshold = std::min(getDposTotalAddressCount(), COMMITTEE_SIZE);
    }

    try {
      v->validate(dpos_votes_count, total_votes_count, sortition_threshold_);
    } catch (const std::logic_error &e) {
      LOG(log_wr_) << e.what();
      return false;
    }
    return true;
  }
```

Comment (JakubFornadel) [reply libraries/core_libs/consensus/src/pbft/pbft_manager.cpp:680]: definitely easier to read :)

**Review by JakubFornadel — APPROVED** [review]

**Review by kstdl — APPROVED** [review]

**Pull request merged**